portal-web/docroot/html/taglib/aui/input/page.jsp

```diff
@@ -25,39 +25,118 @@
 </c:if>
 
 <liferay-util:buffer
-	var="labelContent"
+	var="toggleSwitchLabelContent"
 >
 	<c:if test="<%= Validator.isNotNull(label) %>">
-		<c:if test='<%= type.equals("toggle-switch") %>'>
-			<span class="toggle-switch-label">
+		<span class="toggle-switch-label">
+			<liferay-ui:message key="<%= label %>" localizeKey="<%= localizeLabel %>" />
+		</span>
+
+		<c:if test="<%= required && showRequiredLabel %>">
+			<span class="toggle-switch-text">
+				<liferay-ui:message key="required" />
+
+				<aui:icon cssClass="reference-mark text-warning" image="asterisk" markupView="lexicon" />
+			</span>
 		</c:if>
+	</c:if>
+</liferay-util:buffer>
 
-		<liferay-ui:message key="<%= label %>" localizeKey="<%= localizeLabel %>" />
+<liferay-util:buffer
+	var="toggleSwitchStartContent"
+>
+	<label <%= labelTag %>>
+		<c:if test='<%= inlineLabel.equals("left") %>'>
+			<%= toggleSwitchLabelContent %>
+		</c:if>
 
-		<c:if test='<%= type.equals("toggle-switch") %>'>
+		<span aria-hidden="true" class="toggle-switch-check-bar">
+</liferay-util:buffer>
+
+<liferay-util:buffer
+	var="toggleSwitchEndContent"
+>
+
+			<%
+			String buttonIconOff = (String)dynamicAttributes.get("buttonIconOff");
+			String buttonIconOn = (String)dynamicAttributes.get("buttonIconOn");
+			String iconOff = (String)dynamicAttributes.get("iconOff");
+			String iconOn = (String)dynamicAttributes.get("iconOn");
+			String labelOff = (String)dynamicAttributes.get("labelOff");
+			String labelOn = (String)dynamicAttributes.get("labelOn");
+
+			if (localizeLabel) {
+				if (Validator.isNotNull(labelOff)) {
+					labelOff = LanguageUtil.get(resourceBundle, labelOff);
+				}
+
+				if (Validator.isNotNull(labelOn)) {
+					labelOn = LanguageUtil.get(resourceBundle, labelOn);
+				}
```

Comment:
We can do this inline
```
<span 
    class="toggle-switch-handle"
    data-label-off="<%= (labelOff != null) ? HtmlUtil.escapeAttribute(LanguageUtil.get(resourceBundle, labelOff)) : StringPool.BLANK %>"
    data-label-off=...
```

Reply:
What do we do about `localizeLabel` here? How it currently works in master is it outputs YES or NO if `localizeLabel` is `false` or if `labelOff`/`labelOn` is `null`. This pr gets rid of that and outputs nothing instead. I guess it doesn't make sense not to output anything if there is a value for `labelOff` and `labelOn` just want to double check.

Reply:
It would be a very weird edge case of somebody set values for `labelOff` and `labelOn`, but then set `localizeLabel` to false. `localizeLabel` in general seems like a very unnecessary part of API. If you set `labelOff`, show localized value, if not, show default. You don't need anything else.

Maybe there is a case that I'm not seeing that would make `localizeLabel` valuable. If that's the case, then yeah, we should keep this as it is.